### PR TITLE
Uncontrolled input  component updates

### PR DESCRIPTION
**Pull request by lukemojo**
Closes TD-2588

## Changes

Add props to RUI components to make inputs optionally controlled or uncontrolled.

This pull request will allow us to use each input in an uncontrolled manner for new validated form components. The name field allows the validation library to manage the input fields.

- Add defaultValues/defaultChecked and a name prop to the following components
- Checkbox
- RadioGroup
- TextField
- Switch
- Autocomplete

## UI/UX

No changes

## Testing notes

View the uncontrolled stories in storybook. All components should function as normal and the initial render should render the defaultValue.

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] Branch has been run in docker.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/SwitchField/index.ts
Comment (Skoob1905) [inline src/SwitchField/index.ts:4]: add some docs here

File: src/RadioButtons/index.ts
Comment (Skoob1905) [inline src/RadioButtons/index.ts:6]: add couple docs here

File: src/Checkbox/index.ts
Comment (Skoob1905) [inline src/Checkbox/index.ts:8]: add some docs here

**Review by Skoob1905 — CHANGES_REQUESTED** [review]

File: src/Autocomplete/Autocomplete.stories.tsx
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.stories.tsx:103]:
```suggestion
// Define the uncontrolled story
```

File: src/SwitchField/SwitchField.test.jsx
Comment (mattcorner) [inline src/SwitchField/SwitchField.test.jsx:31]: Your tests seem to either check the rendered value given a defaultValue, or in this case check the onChange handler after a user interaction. I suggest we need both for all the touched components in this PR.

File: src/Autocomplete/Autocomplete.stories.tsx
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.stories.tsx:127]: Do you need the UncontrolledTemplate wrapper in all these stories, or can you just pass the component in directly? It just seems we are removing all state etc which ends up just being the underlying component.

The TextField story is a clear example of this.

```suggestion
  render: Autocomplete
```

**Comment by mattcorner** [comment]
image

I know its not related to this PR, but I saw this in the storybook logs. Can we just delete the right hand side of this code please? We never hit it as vitejs tells us.

**Review by mattcorner — CHANGES_REQUESTED** [review]
Generally looks good. 

Up to you, but it might be a good idea to do a prerelease of RUI from this branch and install it in a couple apps. We're changing some pretty fundamental components here so just to make sure we haven't broken anything.

File: src/SwitchField/SwitchField.test.jsx
Comment (lukemojo) [reply src/SwitchField/SwitchField.test.jsx]: In this instance I was only checking the onChange to infer the default value but have updated it to just check the default value like the checkbox. The onChange events for each component haven't been touched and will just reflect the change event  as per existing behaviour.

**Review by Skoob1905 — APPROVED** [review]
- **Autocomplete:** I uploaded a carmaker test run both to local and devel and then viewed an autocomplete component behaviour is the same.
- **Checkbox:** I changed the road filters in general and the behaviour is matched.
- **LabelSelector:** I uploaded a road with a label called test and I can filter in both devel and locally
- **Radio:** Formgroups are conditionally rendered when using radio buttons, behaviour is the same.
- **Switchfield:** I can successfully change between admin and standard in VIRTO.ID and 
- **Textfield:** Tested this using the search filter, Germany yielded some results and "Germanu" returned no results found for both local and devel

These changes haven't affected any current vital functionality in the app. The usage of using the defaultValues needs to be tested at a later date as I don't have the "middleware" yet to convert the map into a component, but this will come in a later PR.

**Comment by Skoob1905** [comment]
Probably best to just get Matts seal of approval on this before merging this in.

**Comment by lukemojo** [comment]
Please note I've now added a name prop to autocomplete

**Comment by mattcorner** [comment]
Conflict to resolve, then go ahead and merge 👍

**Review by mattcorner — CHANGES_REQUESTED** [review]

**Review by mattcorner — APPROVED** [review]
lgtm

**Pull request merged**
